### PR TITLE
ROS#160: Moved `critical-section` to Aerugo, removed from HAL

### PR DESCRIPTION
**Pull request by SteelPh0enix**

### CODE REVIEW

File: src/boolean_condition.rs [hunk 1]
```diff
@@ -11,10 +11,10 @@ pub use self::boolean_condition_storage::BooleanConditionStorage;
 
 use crate::aerugo::{Aerugo, AERUGO};
 use crate::api::{InitError, SystemApi};
-use crate::arch::Mutex;
 use crate::data_provider::DataProvider;
 use crate::internal_list::InternalList;
 use crate::tasklet::TaskletPtr;
+use crate::Mutex;
```
Comment (Glamhoth) [inline src/boolean_condition.rs:17]:
```suggestion
use crate::mutex::Mutex;
```

File: src/executor.rs [hunk 2]
```diff
@@ -9,8 +9,8 @@ use heapless::binary_heap::{BinaryHeap, Max};
 
 use crate::aerugo::{Aerugo, AERUGO};
 use crate::api::{RuntimeApi, RuntimeError, SystemApi};
-use crate::arch::Mutex;
 use crate::tasklet::{TaskletPtr, TaskletStatus};
+use crate::Mutex;
```
Comment (Glamhoth) [inline src/executor.rs:13]:
```suggestion
use crate::mutex::Mutex;
```

File: src/tasklet.rs [hunk 3]
```diff
@@ -28,10 +28,10 @@ use core::cell::{OnceCell, UnsafeCell};
 
 use crate::aerugo::AERUGO;
 use crate::api::{InitError, RuntimeApi};
-use crate::arch::Mutex;
 use crate::boolean_condition::BooleanConditionSet;
 use crate::data_provider::DataProvider;
 use crate::Instant;
+use crate::Mutex;
```
Comment (Glamhoth) [inline src/tasklet.rs:34]:
```suggestion
use crate::mutex::Mutex;
```

File: arch/cortex-m/aerugo-samv71-hal/src/hal.rs [hunk 4]
```diff
@@ -23,6 +22,8 @@ use samv71_hal::watchdog::{Watchdog, WatchdogConfig};
 /// Safety of this cell is managed by HAL instead, guaranteeing that undefined behavior will not occur.
 static mut HAL_SYSTEM_PERIPHERALS: Option<SystemPeripherals> = None;
 
+/// Global "restore state" that's used to manage
```
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply arch/cortex-m/aerugo-samv71-hal/src/hal.rs]: removed

File: src/boolean_condition.rs (hunk 1 above)
Comment (SteelPh0enix) [reply src/boolean_condition.rs]: renamed

File: src/executor.rs (hunk 2 above)
Comment (SteelPh0enix) [reply src/executor.rs]: renamed

**Review by Glamhoth — APPROVED** [review]

**Pull request merged**